prebuilt-tasks/src/test/java/com/redhat/parodos/tasks/tibco/TibcoWorkFlowTaskTest.java

```diff
@@ -0,0 +1,75 @@
+// @formatter:off
+package com.redhat.parodos.tasks.tibco;
+
+import com.redhat.parodos.tasks.tibco.TibcoWorkFlowTask;
+import com.redhat.parodos.tasks.tibco.Tibjms;
+import com.redhat.parodos.workflow.context.WorkContextDelegate;
+import com.redhat.parodos.workflow.exception.MissingParameterException;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import javax.jms.JMSException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+public class TibcoWorkFlowTaskTest {
+
+	String caFile = "/cafile";
+	String url = "ssl://localhost:7222";
+	String username = "username";
+	String passowrd = "password";
+	String topic = "topic";
+	String message = "test message";
+	Tibjms tibjms = Mockito.mock(Tibjms.class);
+	TibcoWorkFlowTask task = new TibcoWorkFlowTask(tibjms, url, caFile, username, passowrd);
+
+	@Before
+	public void before() {
+		task.setBeanName("Test");
+	}
+	@Test
+	public void executeMissingParameter() {
+		WorkContext ctx = getWorkContext(false);
+		WorkReport result = task.execute(ctx);
+		assertEquals(WorkStatus.FAILED, result.getStatus());
+		assertEquals(MissingParameterException.class, result.getError().getClass());
+	}
+
+	@Test
+	public void executeErrorInTibco() throws JMSException {
+		WorkContext ctx = getWorkContext(true);
+		doThrow(JMSException.class).when(tibjms).sendMessage(any(),any(),any(),any(),any(),any());
+		WorkReport result = task.execute(ctx);
+		assertEquals(WorkStatus.FAILED, result.getStatus());
+		assertEquals(JMSException.class, result.getError().getClass());
+	}
+
+	@Test
+	public void executeSuccess() throws JMSException {
+		WorkContext ctx = getWorkContext(true);
+		WorkReport result = task.execute(ctx);
+		assertEquals(WorkStatus.COMPLETED, result.getStatus());
+		verify(tibjms, times(1)).sendMessage(url, caFile, username, passowrd, topic, message);
+	}
+
+	private WorkContext getWorkContext(boolean withParams) {
+		WorkContext ctx = new WorkContext();
+		Map map;
```

Comment:
no need for type for the Map?

Reply:
no

Reply:
+1 Moti
In truth, this generates the warning "Raw use of parameterized class".
You must provide the proper generic value inside diamonds.

Reply:
this is standard java usage. It makes code simpler and less type independent where possible. So when you change the map type you want to use you change only where it is created and only where necessary. That's the whole idea of interfaces and generics (templates). But again...just for merging the PR I will change it. BTW...intellij does that automatically. Not my idea.

Reply:
I don't see a warning